pkg/client/watch.go

```diff
@@ -33,10 +33,14 @@ func NewWithWatch(config *rest.Config, options Options) (WithWatch, error) {
 	if err != nil {
 		return nil, err
 	}
-	dynamicClient, err := dynamic.NewForConfig(config)
+
+	var dynamicClient dynamic.Interface
+
+	dynamicClient, err = dynamic.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
+
 	return &watchingClient{client: client, dynamic: dynamicClient}, nil
```

Comment:
Need to see what, if any, adjustments need to be made to work with * or cluster-specific watches.